core/legacy/regular-update.js

```diff
@@ -1,93 +1,62 @@
-import requestModule from 'request'
```

Comment:
This file is the last file left in `core/legacy/` and it has no unit tests although it is not _completely_ untested because the service tests for all the services in the title exercise this code.

My instinct is that having migrated this to got and async/await we should move this out of `core/legacy/` and write a proper test suite for it. Before I go ahead with that, it would be useful to understand why this is in legacy. Maybe @paulmelnikow  can chime in on this one. Does this sufficiently modernise it, or is there another reason we want to ditch this?

Reply:
I'm going to go ahead and merge this so we can get #7214 resolved, but it would be useful to get some input on this anyway and I will follow up